### ISSUE TITLE
Request: Keyboard navigation in md-list

### ISSUE THREAD

**Issue by vipdorian** [angular/material2#941]
To be able to use arrow keys to navigate through md-list or md-data-table or md-grid-list

As said in http://webaim.org/techniques/keyboard/ , in the section "Navigate to most elements" : 
- Keyboard focus indicators must be present.
- Navigation order should be logical and intuitive.

In material-angular (1.1.0-rc.5), arrow keys does not select or change focus in md-list. The issue was closed in material-angular as part of ‘Surge Focus on Material 2' efforts. Requesting this feature to be included in material-angular-2.

Use-case 1 : Pharmacy Software, need to navigate through a md-list of prescriptions with up/down. 
<img width="1142" alt="screen shot 2016-08-02 at 00 25 02" src="https://cloud.githubusercontent.com/assets/10145164/17317048/9c43d9c8-5847-11e6-93b9-b37415b453a1.png">

Use-case 2 : Google drive, you can navigate through files with arrow keys.
<img width="1100" alt="screen shot 2016-08-02 at 00 22 59" src="https://cloud.githubusercontent.com/assets/10145164/17317028/6c124dc0-5847-11e6-8560-007d4da413eb.png">

Use-case 3 : Google mail, you can navigate through emails with arrow keys.

Motivation : Most pharmacists in Quebec are using an old C# pharmacy software which is ugly but fast. If I can tell them that Material-Angular2 is good-looking AND fast, it might be a game-changer. (keyboard-friendly == fast)

**Comment by jelbourn** [comment]
As mentioned in #1999, this is something that is more appropriate for `md-action-list` and `md-selection-list` once those are implemented.
